src/components/LectureSelect.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "./ui/select"
+import { lectures } from "../data/lectures"
+import { navigate } from "gatsby"
+import slugify from "slugify"
+
+const handleSelectChange = (val: string) => {
+  console.log("e.target.value", val)
+  const selectedIndex = parseInt(val) || 0
+  navigate(`/lectures/${slugify(lectures[selectedIndex].section)}`)
```

Comment:
`navigate` is not what we want here. We just want to update the url. 

As of now you add it as a actual "resource" to the url like this:

`https://graypaper-website.vercel.app/lectures/1.1-Nomenclature/`

suggesting there is an actual page `1.1-Nomenclature`.

However what we maybe rather one is adding it as a [url search parameter](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). This indicates it's a value used on the the site `/lectures`

`https://graypaper-website.vercel.app/lectures?chapter=1.1-Nomenclature/`

Either way, both works fine. 

I think this would work. However you need to retrieve the slug from the url search param instead from the full url path.
```js
navigate(`/lectures?chapter=${slugify(lectures[selectedIndex].section)}`
```

If you wanto to update the whole url path without navigating happening you should use:


```js
history.pushState({},"URL Rewrite Example","https://stackoverflow.com/example")
```

https://stackoverflow.com/questions/8560617/how-to-change-url-in-browser-without-navigating-away-from-page

Reply:
Yep will either do that or really create dynamic pages. But url Parsons is probably quicker.